packages/core/src/middleware/koa-log.ts

```diff
@@ -60,18 +81,19 @@ const initLogger = (basePayload?: Readonly<BaseLogPayload>) => {
 };
 /* eslint-enable @silverhand/fp/no-mutation */
 
-export default function koaLog<StateT, ContextT, ResponseBodyT>(): MiddlewareType<
+export default function koaLog<
   StateT,
-  WithLogContext<ContextT>,
+  ContextT extends IRouterParamContext,
   ResponseBodyT
-> {
+>(): MiddlewareType<StateT, WithLogContext<ContextT>, ResponseBodyT> {
   return async (ctx, next) => {
     const {
       ip,
       headers: { 'user-agent': userAgent },
     } = ctx.request;
 
     const logger = initLogger({ result: LogResult.Success, ip, userAgent });
+    ctx.logSession = logger.logSession;
```

Comment:
Is it possible to reuse `ctx.log`?

Reply:
cannot.
`koa-log-session` middleware does not know using which `type`, the first parameter of `ctx.log(type, payload)`.
seems that we still need to add another function for `koa-log-session` middleware, or `ctx.log` function has to be more complex.

Reply:
add some methods like "addBreadCrumb"? more generic function.

Reply:
refactored: merged SessionLogPayload into BaseLogPayload. PTAL